### PR TITLE
Fig changed to Compose

### PR DESCRIPTION
**Pull request by Robaum**
Fig changed to Docker Compose. I also did some extra minor changes in the config files and the Dockerfile.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,7 +9,7 @@ Thumbs.db
 # ==========================
 .nodemonignore
 .sass-cache/
-npm-debug.log
+npm-debug.log*
```
Comment (ilanbiala) [inline .gitignore]: What is the `*` for?

File: README.md [hunk 2]
```diff
@@ -101,12 +101,12 @@ $ grunt test:client
 
 ## Development and deployment With Docker
 
-* Install [Docker](http://www.docker.com/)
-* Install [Fig](https://github.com/orchardup/fig)
+* Install [Docker](https://docs.docker.com/installation/#installation)
+* Install [Compose](http://docs.docker.com/compose/install/)
```
Comment (ilanbiala) [inline README.md]: `https` works here too.

File: config/env/development.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 module.exports = {
 	db: {
-		uri: 'mongodb://localhost/mean-dev',
+    uri: process.env.MONGOHQ_URL || process.env.MONGOLAB_URI || 'mongodb://' + (process.env.DB_1_PORT_27017_TCP_ADDR || 'localhost') + '/mean-dev',
```
Comment (ilanbiala) [inline config/env/development.js]: wrong indentation

File: config/env/test.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 module.exports = {
 	db: {
-		uri: 'mongodb://localhost/mean-test',
+    uri: process.env.MONGOHQ_URL || process.env.MONGOLAB_URI || 'mongodb://' + (process.env.DB_1_PORT_27017_TCP_ADDR || 'localhost') + '/mean-test',
```
Comment (ilanbiala) [inline config/env/test.js]: wrong indentation

File: docker-compose.yml [hunk 5]
```diff
@@ -4,9 +4,9 @@ web:
...
    - "3000:3000"
-  environment: 
+  environment:
    NODE_ENV: development
 db:
   image: mongo
-  ports: 
-   - "27017:27017"
+  ports:
+   - "27017:27017"
```
Comment (ilanbiala) [inline docker-compose.yml]: missing newline at end of file

File: Dockerfile [hunk 6]
```diff
@@ -20,7 +20,7 @@ RUN bower install --config.interactive=false --allow-root
 # Make everything available for start
 ADD . /home/mean
 
-# currently only works for development
+# Set development enviornment as default
```
Comment (pgrodrigues) [inline Dockerfile]: typo in "environment"

File: .gitignore (hunk 1 above)
Comment (Robaum) [inline .gitignore]: I am not 100% sure now npm handles debug logs but when testing I had 2 different logs and their names where (for example) npm-debug.log.12345. The `*` is to ignore all the extra characters the log might have. Might be something that just came up in my implementation but I think is a nice addition.

File: Dockerfile (hunk 6 above)
Comment (Robaum) [inline Dockerfile]: Thank you :)

File: README.md (hunk 2 above)
Comment (Robaum) [inline README.md]: It does. Do you prefer to have `https`?

File: config/env/test.js [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 module.exports = {
 	db: {
-		uri: 'mongodb://localhost/mean-test',
+    		uri: process.env.MONGOHQ_URL || process.env.MONGOLAB_URI || 'mongodb://' + (process.env.DB_1_PORT_27017_TCP_ADDR || 'localhost') + '/mean-test',
```
Comment (ilanbiala) [inline config/env/test.js]: What's with the spacing before the text here and in `config/env/secure.js:6`?

Comment (Robaum) [inline config/env/test.js]: What is wrong with the spacing? Maybe my text editor did some funny business I can't see.

Comment (ilanbiala) [inline config/env/test.js]: There are either tabs or spaces but unnecessary extra spaces. GitHub shows them, so I would just delete the line and press enter and it should be better.

**Comment by ilanbiala** [comment]
When you make any change, can you just squash so it's easier to keep track of what still needs to be tidied up?

File: Dockerfile [hunk 8]
```diff
@@ -1,4 +1,4 @@
-FROM dockerfile/nodejs
+FROM node:latest
```
Comment (isotnikov) [inline Dockerfile]: I guess need to specify node version instead of using latest node docker image.

Comment (Robaum) [inline Dockerfile]: You are right. Do you have any recommendations for the version that should be used?

**Comment by Robaum** [comment]
Sorry for my absense but I am back. Commits where squashed and files updated for no extra spaces.

**Comment by ilanbiala** [comment]
@Robaum cool. @isotnikov. @pgrodrigues, is there anything missing from this? @lirantal LGTY?

**Comment by Robaum** [comment]
@ilanbiala Sorry I didn't understood. You want to me do a pull request against the latest master?

File: Dockerfile (hunk 8 above)
Comment (isotnikov) [inline Dockerfile]: Try 'node:0.10.39'

**Comment by isotnikov** [comment]
@ilanbiala looks like all ready. @Robaum sorry for delay, 'node:10' in the Dockerfile is fine.

**Comment by Robaum** [comment]
Ok I got it. You want me to do the pull request against branch 0.4.0 and not master. Do I have to make a new pull request or can I change this one?

**Comment by lirantal** [comment]
@Robaum yep, great.
Once you do the PR please close this one.

**Comment by Robaum** [comment]
Closed this PR. New PR #611

**Pull request closed without merging**